src/main/java/org/apache/xml/security/keys/content/AgreementMethodImpl.java

```diff
@@ -0,0 +1,324 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements. See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership. The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at
+ * <p>
+ * http://www.apache.org/licenses/LICENSE-2.0
+ * <p>
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.xml.security.keys.content;
+
+import org.apache.xml.security.encryption.AgreementMethod;
+import org.apache.xml.security.encryption.KeyDerivationMethod;
+import org.apache.xml.security.encryption.XMLEncryptionException;
+import org.apache.xml.security.encryption.params.ConcatKeyDerivationParameter;
+import org.apache.xml.security.encryption.params.KeyAgreementParameterSpec;
+import org.apache.xml.security.exceptions.XMLSecurityException;
+import org.apache.xml.security.keys.OriginatorKeyInfo;
+import org.apache.xml.security.keys.RecipientKeyInfo;
+import org.apache.xml.security.keys.derivedKey.ConcatKDFParamsImpl;
+import org.apache.xml.security.keys.derivedKey.KeyDerivationMethodImpl;
+import org.apache.xml.security.utils.*;
+import org.w3c.dom.Document;
+import org.w3c.dom.Element;
+
+import java.lang.System.Logger;
+import java.lang.System.Logger.Level;
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.security.PublicKey;
+import java.util.Iterator;
+import java.util.LinkedList;
+import java.util.List;
+
+
+/**
+ * The implementation of the AgreementMethod interface. The element contains a information about
+ * the key agreement algorithm for deriving the encryption key.
+ *
+ */
+public class AgreementMethodImpl extends EncryptionElementProxy implements KeyInfoContent, AgreementMethod {
```

Comment:
Move this and other encryption specific classes to `org.apache.xml.security.encryption` package.